app/src/main/java/org/oppia/app/player/state/ImageRegionSelectionInteractionView.kt

```diff
@@ -0,0 +1,98 @@
+package org.oppia.app.player.state
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.databinding.BindingAdapter
+import androidx.fragment.app.FragmentManager
+import org.oppia.app.fragment.InjectableFragment
+import org.oppia.app.model.ImageWithRegions
+import org.oppia.app.utility.ClickableAreasImage
+import org.oppia.app.utility.OnClickableAreaClickedListener
+import org.oppia.util.accessibility.CustomAccessibilityManager
+import org.oppia.util.gcsresource.DefaultResourceBucketName
+import org.oppia.util.parser.ExplorationHtmlParserEntityType
+import javax.inject.Inject
+
+/**
+ * A custom [AppCompatImageView] with a list of [LabeledRegion] to work with
+ * [ClickableAreasImage].
+ */
+class ImageRegionSelectionInteractionView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : AppCompatImageView(context, attrs, defStyleAttr) {
+
+  private var isAccessibilityEnabled: Boolean = false
+  private var clickableAreas: List<ImageWithRegions.LabeledRegion> = emptyList()
+
+  @Inject
+  lateinit var accessibilityManager: CustomAccessibilityManager
+
+  @Inject
+  @field:ExplorationHtmlParserEntityType
+  lateinit var entityType: String
+
+  @Inject
+  @field:DefaultResourceBucketName
+  lateinit var resourceBucketName: String
+
+  private lateinit var entityId: String
+
+  fun setEntityId(entityId: String) {
+    this.entityId = entityId
+  }
+
+  fun setClickableAreas(clickableAreas: List<ImageWithRegions.LabeledRegion>) {
+    this.clickableAreas = clickableAreas
+  }
+
+  fun getClickableAreas(): List<ImageWithRegions.LabeledRegion> {
+    return clickableAreas
+  }
+
+  fun isAccessibilityEnabled(): Boolean {
+    return isAccessibilityEnabled
+  }
+
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    FragmentManager.findFragment<InjectableFragment>(this).createViewComponent(this).inject(this)
+    isAccessibilityEnabled = accessibilityManager.isScreenReaderEnabled()
+  }
+}
+
+/** Bind ItemTouchHelperSimpleCallback with RecyclerView for a [DragDropSortInteractionView] via data-binding. */
+@BindingAdapter(value = ["onRegionClicked", "overlayView"], requireAll = false)
+fun setRegionClickToImageView(
+  imageRegionSelectionInteractionView: ImageRegionSelectionInteractionView,
+  onClickableAreaClickedListener: OnClickableAreaClickedListener,
+  parentView: FrameLayout
+) {
+  val area = ClickableAreasImage(
+    imageRegionSelectionInteractionView,
+    parentView,
```

Comment:
I don't think so we can rely on an automatic one because there would be two values to bind and in the article as well there is binding adapter if wish to bind two values.